packages/lib/services/rest/routes/notes.ts

```diff
@@ -223,7 +226,7 @@ const isValidUrl = (url: string, isDataUrl: boolean, urlProtocol?: string, allow
 	return isAllowedProtocol;
 };
 
-export async function downloadMediaFile(url: string, fetchOptions?: FetchOptions, allowedProtocols?: string[]) {
+export async function downloadMediaFile(url: string, downloadController: DownloadController, fetchOptions?: FetchOptions, allowedProtocols?: string[]) {
```

Comment:
Please make the downloadController optional.

I know we have this rule to avoid optional parameters, but when it means having to create a dummy object on each call it's better to make it optional

Reply:
Also why do we need this dummy controller? Isn't it possible to make the code work with a simple `null` controller? I guess there will be a few null checks here and there. 

Especially since `fetchBlob` is heavily used for sync it's better to avoid creating unnecessary objects on each call.

Reply:
And if it's optional, make it part of the "fetchOptions" object

Reply:
I move the downloadController to be a property of the other options object and instead of creating a dummy class I'm checking before using downloadController like you suggested.